internal/cel/transformer.go

```diff
@@ -0,0 +1,128 @@
+package transformer
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+	"sort"
+
+	"github.com/google/cel-go/common"
+	"github.com/google/cel-go/common/ast"
+	"github.com/google/cel-go/parser"
+)
+
+func parseExpression(expression string) (*ast.AST, error) {
+	prsr, err := parser.NewParser()
+	if err != nil {
+		return nil, err
+	}
+
+	p, iss := prsr.Parse(common.NewTextSource(expression))
+	if len(iss.GetErrors()) > 0 {
+		return nil, errors.New("got errors parsing")
+	}
+
+	return p, nil
+}
+
+func TransformCounterVariable(expression string) (*string, error) {
```

Comment:
as this is very limitador v1 to v2 specific, a code comment could be useful here